dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt

```diff
@@ -295,10 +295,152 @@ internal class RumViewScopeTest {
+        // Given
+        val fakeViewEventType = forge.aValueFrom(RumViewScope.RumViewType::class.java)
+
+        // When
+        testedScope = RumViewScope(
+            mockParentScope,
+            fakeKey,
+            fakeName,
+            fakeEventTime,
+            fakeAttributes,
+            mockDetector,
+            mockCpuVitalMonitor,
+            mockMemoryVitalMonitor,
+            mockFrameRateVitalMonitor,
+            mockTimeProvider,
+            mockRumEventSourceProvider,
+            mockBuildSdkVersionProvider,
+            mockViewUpdatePredicate,
+            type = fakeViewEventType
+        )
+
+        // Then
+        assertThat(GlobalRum.getRumContext().viewType).isEqualTo(fakeViewEventType)
+    }
+
+    @Test
+    fun `𝕄 update the context with viewType NONE W handleEvent(StopView)`(
+        forge: Forge
+    ) {
+        // Given
+        val attributes = forge.exhaustiveAttributes(excludedKeys = fakeAttributes.keys)
+        val expectedAttributes = mutableMapOf<String, Any?>()
+        expectedAttributes.putAll(fakeAttributes)
+        expectedAttributes.putAll(attributes)
+
+        // When
+        testedScope.handleEvent(
+            RumRawEvent.StopView(fakeKey, attributes),
+            mockWriter
+        )
+
+        // Then
+        assertThat(GlobalRum.getRumContext().viewType).isEqualTo(RumViewScope.RumViewType.NONE)
+    }
+
+    @Test
+    fun `𝕄 not update the context with viewType NONE W handleEvent(StopView) { unknown key }`(
+        forge: Forge
+    ) {
+        // Given
+        val attributes = forge.exhaustiveAttributes(excludedKeys = fakeAttributes.keys)
+        val expectedAttributes = mutableMapOf<String, Any?>()
+        expectedAttributes.putAll(fakeAttributes)
+        expectedAttributes.putAll(attributes)
+
+        // When
+        testedScope.handleEvent(
+            RumRawEvent.StopView(forge.anAlphabeticalString() + fakeKey, attributes),
+            mockWriter
+        )
+
+        // Then
+        assertThat(GlobalRum.getRumContext().viewType)
+            .isEqualTo(RumViewScope.RumViewType.FOREGROUND)
+    }
+
     // endregion
 
     // region View
 
+    @Test
+    fun `𝕄 not update the viewType to NONE W handleEvent(StartView) { on active view }`(
+        @StringForgery key: String,
+        @StringForgery name: String
+    ) {
+        // Given
+        testedScope.handleEvent(
+            RumRawEvent.StopView(fakeKey, emptyMap()),
+            mockWriter
+        )
+        RumViewScope(
+            mockParentScope,
+            key,
+            name,
+            fakeEventTime,
+            fakeAttributes,
+            mockDetector,
+            mockCpuVitalMonitor,
+            mockMemoryVitalMonitor,
+            mockFrameRateVitalMonitor,
+            mockTimeProvider,
+            mockRumEventSourceProvider,
+            mockBuildSdkVersionProvider,
+            mockViewUpdatePredicate
```

Comment:
minor: we maybe need to write parameterized test here, by checking both `FOREGROUND` and `BACKGROUND` values, same for the test below.